src/common.rkt

```diff
@@ -104,6 +100,72 @@
   (check-true (set-disjoint? '() '()))
   (check-false (set-disjoint? '(a b c) '(a))))
 
+(define (subsequence? v l)
+  (or
+   (empty? v)
+   (let ([v* (member (first v) l)])
+     (and v* (subsequence? (rest v) v*)))))
+
+(module+ test
+  (define l (range 10))
+  (check-true (subsequence? empty empty))
+  (check-true (subsequence? empty l))
+  (check-true (subsequence? '(1) l))
+  (check-true (subsequence? '(1 2) l))
+  (check-true (subsequence? '(1 3 5 7 9) l))
+  (check-true (subsequence? '(1 2 5 8) l))
+  (check-false (subsequence? '(x y) l))
+  (check-false (subsequence? '(1 2 10) l)))
+
+(define (list-ref* l p)
+  (let loop ([l l] [p p] [i 0])
+    (if (or (empty? p) (empty? l))
+        empty
+        (let* ([j (first p)]
+               [k (- j i)]
+               [l* (drop l k)])
+          (cons (first l*) (loop (rest l*) (rest p) (+ i k 1)))))))
+
+(module+ test
+  (define m '(a b c d e f g))
+  (check-equal? (list-ref* empty empty) empty)
+  (check-equal? (list-ref* m empty) empty)
+  (check-equal? (list-ref* m '(1)) '(b))
+  (check-equal? (list-ref* m '(0 2 4 6)) '(a c e g))
+  (check-equal? (list-ref* m '(0 2 3 5 6)) '(a c d f g)))
+
+(define (list-set* l p v)
```

Comment:
Like `list-ref` I would just inline `(map (curry list-set l) p v)` and call it a day. Yes, it's O(n^2) instead of O(n) but in all cases we can imagine `n` will be, like, 15.

Reply:
After trying `(map (curry list-set l) p v)`, I realized it actually won't work. That gives a list of lists where each element is `l` with the value at `p` changed to `v`. We need to be mapping over `l` and have some method of deciding whether to substitute values from `sorted` while we do it, which I found to be kinda tricky. Before landing on `list-set*` (which btw is just a slightly modified version of `list-set-multiple` from "symmetry.rkt" when it existed), I tried several other ways of doing it, but I couldn't figure how to do it inline using just the standard library. Any ideas?